association/association_analysis.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import pandas as pd
+from mlxtend.frequent_patterns import apriori
+from mlxtend.frequent_patterns import association_rules
+
+
+def add_item(hmap, key, new_item):
+    if key in hmap:
+        hmap[key].append(new_item)
+    else:
+        hmap[key] = [new_item]
+
+
+def create_hashmap(df, hmap):
+    for i, cols in df.iterrows():
+        key = df['transactionId'][i]
+        add_item(hmap, key, i)
```

Comment:
I propose to remove argument `hmap` from this function, and create the dictionary directly inside the function:

```py
def create_tx_map(df):
    # (k, v): (transaction_id, list(index))
    tx_map = {}
    for i, cols in df.iterrows():
        key = df['transactionId'][i]
        tx_map.setdefault(key, [])
        tx_map.append(i)
    return tx_map
```

And then, you can also remove the function `add_item(hmap, key, new_item)`, which has already been implemented by the `setdefault` function, as you can see above. See also _[How to add multiple values to a dictionary key in python?](https://stackoverflow.com/questions/20585920/how-to-add-multiple-values-to-a-dictionary-key-in-python)_